src/circuits/blake2s/blake2s_comp.hpp

```diff
@@ -0,0 +1,74 @@
+#ifndef __ZETH_BLAKE2S_HASH_HPP__
+#define __ZETH_BLAKE2S_HASH_HPP__
+
+#include "circuits/circuits-util.hpp"
+#include "circuits/simple_gadgets.hpp"
+#include "g_primitive.hpp"
+#include "types/bits.hpp"
+#include "util.hpp"
+
+#include <libsnark/gadgetlib1/gadget.hpp>
+#include <libsnark/gadgetlib1/gadgets/basic_gadgets.hpp>
+#include <libsnark/gadgetlib1/gadgets/hashes/hash_io.hpp>
+#include <math.h>
+
+namespace libzeth
+{
+
+const size_t BLAKE2s_digest_size = 256;
+const size_t BLAKE2s_block_size = 512;
+
+template<typename FieldT>
+class BLAKE2s_256_comp : public libsnark::gadget<FieldT>
+{
+private:
+    static const int rounds = 10;
+    std::array<std::array<FieldT, 32>, 8> h;
+    std::array<std::array<FieldT, 32>, 2> t;
+    std::array<libsnark::pb_variable_array<FieldT>, 16> block;
+    std::array<std::array<libsnark::pb_variable_array<FieldT>, 16>, rounds + 1>
+        v;
+    std::array<std::array<libsnark::pb_variable_array<FieldT>, 16>, rounds>
+        v_temp;
+    std::array<libsnark::pb_variable_array<FieldT>, 8> output_bytes;
+    libsnark::block_variable<FieldT> input_block;
+    libsnark::digest_variable<FieldT> output;
+
+    std::array<std::vector<g_primitive<FieldT>>, 10> g_arrays;
+    std::vector<xor_constant_gadget<FieldT>> xor_vector;
+
+    // TODO: Remove ZERO and pass it in the constructor
```

Comment:
Can you pass the ZERO to the constructor?

Reply:
I started doing this in the fix-todos branch (c.f. sha256, joinsplit circuits...) . I intended to do this on that branch once the blake2s PR is finished.

Reply:
As you prefer as long as we do not forget :)

Reply:
Done in fix-todo PR